accumulator_updater/programs/accumulator_updater/src/state/accumulator_input.rs

```diff
@@ -14,26 +14,94 @@ use {
 /// the CPI calling program (e.g. Pyth Oracle)
 ///
 /// TODO: implement custom serialization & set alignment
-#[account]
+#[account(zero_copy)]
+#[derive(Debug, InitSpace)]
 pub struct AccumulatorInput {
     pub header: AccumulatorHeader,
-    pub data:   Vec<u8>,
+    // 10KB - 8 (discriminator) - 514 (header)
+    // TODO: do we want to initialize this to the max size?
+    //   - will lead to more data being passed around for validators
+    pub data:   [u8; 9_718],
 }
 
+//TODO:
+// - implement custom serialization & set alignment
+// - what other fields are needed?
+#[zero_copy]
+#[derive(InitSpace, Debug)]
+pub struct AccumulatorHeader {
+    pub bump:        u8, // 1
+    pub version:     u8, // 1
+    // byte offset of accounts where data starts
+    // e.g. account_info.data[offset + header_len]
+    pub header_len:  u16, // 2
+    /// endpoints of every message.
+    /// ex: [10, 14]
+    /// => msg1 = data[(header_len + 0)..(header_len + 10)]
+    /// => msg2 = data[(header_len + 10)..(header_len + 14)]
+    pub end_offsets: [u16; 255], // 510
+}
+
+impl AccumulatorHeader {
+    // header_size. Keeping same structure as `BatchPriceAttestation` header
+    // HEADER_LEN allows for append-only forward-compatibility for the header.
+    // this is the number of bytes from the beginning of the account_info.data
+    // to the start of the `AccumulatorInput` data.
+    //
+    // *NOTE* this implementation is slightly different
+    // than the `BatchPriceAttestation` header_size
```

Comment:
(fwiw i think this comment would be better if it didn't reference BatchPriceAttestation. The reader probably won't understand how that format works deeply, so the comparison likely won't help)